x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/FunctionRegistry.java

```diff
@@ -221,7 +222,7 @@ public FunctionDefinition resolveFunction(String name) {
 
     public String concreteFunctionName(String alias) {
         String normalized = normalize(alias);
-        return aliases.getOrDefault(normalized, normalized);
+        return aliases.getOrDefault(normalized, alias.toUpperCase(Locale.ROOT));
```

Comment:
I agree with the `alias.toUpperCase()` but I don't see why we normalize the received name. This means we accept `CharLength` as a function name, but if the function name is incorrect (for example `CharLen`) we tell the user what he meant (CHARACTER_LENGTH, CHAR_LENGTH). No mentioning of CHARLENGTH.

Imo, given that it's impossible to guess where to put the underscore in case of a `CharLength` or `CharlEngth`, we should only accept straight names, without trying to normalize. Otherwise, we won't be able to offer a consistent behavior to the user.

Reply:
But if I understand correctly, with your suggestion we'll break the existing "feature" of converting camelCase to CAMEL_CASE.

Reply:
I don't have any objection on this suggestion. It will simplify things and the rule would be as simple as: apply upperCase to the input and try to resolve from the register function names and aliases.